frontend/src/pages/data-sets/(id)/chat/new/(agent)/index.tsx

```diff
@@ -5,17 +5,16 @@ import { NewConversation } from "@/components/conversation-view/new-conversation
 import {useApplicationContext} from "@/lib/use-application-context.ts";
 
 export function NewChat() {
-  const { availableAgents } = useApplicationContext()!;
+  const { availableAgentInstances } = useApplicationContext()!;
   const { agent } = useParams();
 
-  const selectedAgent = availableAgents.find(agentObj => agentObj.key === agent);
+  const selectedAgent = availableAgentInstances.find(agentObj => agentObj.id.toString() === agent);
```

Comment:
Why do we have to .toString() it?

Reply:
This `agent` comes from route `/<agentId>` and it's always string, while API returns number.
But probably it's better to compare integers in JS instead of string so I inverted this logic.